libs/ledger/benchmark/in_memory_storage.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ledger/chain/digest.hpp"
+#include "ledger/chain/transaction.hpp"
+#include "ledger/storage_unit/storage_unit_interface.hpp"
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+
+class InMemoryStorageUnit : public fetch::ledger::StorageUnitInterface
```

Comment:
this seems like it will be the third or so fake storage unit in the codebase, am I right?